urbs.py

```diff
@@ -964,7 +973,17 @@ def res_process_capacity_rule(m, sit, pro):
     return (m.process.loc[sit, pro]['cap-lo'],
             m.cap_pro[sit, pro],
             m.process.loc[sit, pro]['cap-up'])
-
+            
+# used roof area <= roof area
+def res_process_roof_area_rule(m, sit, pro):
+    for p in m.pro_tuples:
+        try:
+            float(m.process.loc[p]['area-per-cap']) 
+            return (0,
```

Comment:
**Wait (continued)**: as pointed out in the comment on the constraint declaration:
```python
def res_process_roof_area_rule(m, sit):
    sum_process_areas = sum(m.cap_pro[p] * m.process.loc[p]['area-per-cap'] 
                            for (s, p) in m.pro_area_tuples 
                            if s == sit)
    return sum_process_areas <= m.site.loc[sit]['area']
```